File: Embarcadero/XE5/ObjectPascal/Forms/PortraitForm.pas
unit PortraitForm;

interface

uses
  System.SysUtils, System.Types, System.UITypes, System.Classes, System.Variants,
  FMX.Types, FMX.Controls, FMX.Forms, FMX.Dialogs, FMX.StdCtrls, FMX.Objects, FMX.MobilePreview;

type
  TPForm = class(TForm)
    ToolBar1: TToolBar;
    Label2: TLabel;
    Image1: TImage;
    procedure FormResize(Sender: TObject);
  private
  public
    { Public declarations }
  end;

var
  PForm: TPForm;

implementation

{$R *.fmx}

uses
LandscapeForm;

procedure TPForm.FormResize(Sender: TObject);
begin
  if (Height < Width) and (Visible) and Assigned(LSForm) then
    LSForm.Show;
end;

end.

File: Embarcadero/XE5/CPP/Forms/PortraitForm.cpp
//---------------------------------------------------------------------------

#include <fmx.h>
#pragma hdrstop

#include "PortraitForm.h"
#include "LandscapeForm.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma resource "*.fmx"
TPForm *PForm;
//---------------------------------------------------------------------------
__fastcall TPForm::TPForm(TComponent* Owner)
	: TForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TPForm::FormResize(TObject *Sender)
{
	if ((Height < Width)  && (Visible) &&  (LSForm != NULL)) {
		LSForm->Show();
	}
}
//---------------------------------------------------------------------------

